src/server/bin.js

```diff
@@ -2,27 +2,53 @@
 
 'use strict'
 
-// Usage: $0 [<address> <address 2>...]
+// Usage: $0 [-libp2pMultiaddr <multiaddr>] [-metricsPort <port>] [--disableMetrics]
 
 /* eslint-disable no-console */
 
+const http = require('http')
+const menoetius = require('menoetius')
+
+const multiaddr = require('multiaddr')
 const Server = require('.')
 
+const argv = require('minimist')(process.argv.slice(2))
+
 async function run () {
-  let addresses = process.argv.slice(2)
-  if (!addresses.length) { addresses = undefined } // use default if none provided
+  const metrics = !(argv.disableMetrics || process.env.DISABLE_METRICS)
+  const metricsPort = argv.metricsPort || argv.mp || process.env.PORT || 8003
+  const libp2pMa = argv.libp2pMultiaddr || argv.lm || process.env.LIBP2PMA || `/ip6/::/tcp/5892/ws`
+  const addresses = [multiaddr(libp2pMa)]
 
-  const server = new Server({ addresses })
+  let metricsServer
 
+  const server = new Server({ addresses, hasMetrics: metrics })
   await server.start()
 
   console.log('server peerID: ', server.libp2p.peerInfo.id.toB58String())
 
   server.peerAddr.forEach((ma) => console.log('listening on %s', ma.toString()))
 
+  if (metrics) {
+    console.log('enabling metrics')
+    metricsServer = http.createServer((req, res) => {
+      if (req.url !== '/metrics') {
+        res.statusCode = 200
+        res.end()
+      }
+    })
+
+    menoetius.instrument(metricsServer)
+
+    metricsServer.listen(metricsPort, '127.0.0.1', () => {
```

Comment:
host should be configurable aswell. maybe just use a multiaddr for metrics aswell and extract host port

Reply:
I wanted to not complicate this for users. But, I guess that they can only use the defaults. I will change that